python/whylogs/api/writer/whylabs.py

```diff
@@ -476,20 +484,79 @@ def get_feature_weights(self, **kwargs: Any) -> Optional[FeatureWeights]:
             self._dataset_id = kwargs.get("dataset_id")
 
         result = self._do_get_feature_weights()
-        feature_weights_set = result.get("segmentWeights")
+        feature_weights_set = result.get("segment_weights")
         metadata = result.get("metadata")
         if feature_weights_set and isinstance(feature_weights_set, list):
             feature_weights = FeatureWeights(weights=feature_weights_set[0]["weights"], metadata=metadata)
             return feature_weights
         return None
 
+    def write_segmented_reference_result_set(self, file: SegmentedResultSet, **kwargs: Any) -> Tuple[bool, str]:
+        """Put segmented reference result set for the specified dataset.
+
+        Parameters
+        ----------
+        file : SegmentedResultSet
+            SegmentedResultSet object representing the segmented reference result set for the specified dataset
+
+        Returns
+        -------
+        Tuple[bool, str]
+        """
+        utc_now = datetime.datetime.now(datetime.timezone.utc)
+
+        files = file.get_writables()
+        partitions = file.partitions
+        if len(partitions) > 1:
+            logger.warning(
+                "SegmentedResultSet contains more than one partition. Only the first partition will be uploaded. "
+            )
+        partition = partitions[0]
+        whylabs_tags = list()
+        for view in files:
+            view_tags = list()
+            dataset_timestamp = view.dataset_timestamp or utc_now
+            if view.partition.id != partition.id:
+                continue
+            _, segment_tags, _ = _generate_segment_tags_metadata(view.segment, view.partition)
+            for segment_tag in segment_tags:
+                tag_key = segment_tag.key.replace("whylogs.tag.", "")
+                tag_value = segment_tag.value
+                view_tags.append({"key": tag_key, "value": tag_value})
+            whylabs_tags.append(view_tags)
+        stamp = dataset_timestamp.timestamp()
```

Comment:
I think we have a utility that does this? Lets factor out in follow up.

Reply:
https://github.com/whylabs/whylogs/issues/1251